File: aat-android/src/main/java/ch/bailu/aat/preferences/map/SolidTrimSize.java
package ch.bailu.aat.preferences.map;

import android.content.Context;

import ch.bailu.aat.R;
import ch.bailu.aat.preferences.SolidIndexList;
import ch.bailu.aat.util.MemSize;

public class SolidTrimSize extends SolidIndexList {

    private static class Entry {
        public final long size;
        public final String text;

        private Entry(long s) {
            size = s;
            text = MemSize.describe(new StringBuilder(), size).toString();
        }
    }




    private static final Entry[] entries = {
            new Entry(16L * MemSize.GB),
            new Entry(8L * MemSize.GB),
            new Entry(4L * MemSize.GB),
            new Entry(2L * MemSize.GB),
            new Entry(1L * MemSize.GB),
            new Entry(500L * MemSize.MB),
            new Entry(200L * MemSize.MB),
            new Entry(100L * MemSize.MB),
            new Entry(50L * MemSize.MB),
    };


    public SolidTrimSize(Context context) {
        super(context, SolidTrimSize.class.getSimpleName());
    }

    @Override
    public int length() {
        return entries.length;
    }


    @Override
    public String getLabel() {
        return getContext().getString(R.string.p_trim_size);
    }

    @Override
    public String getValueAsString(int i) {
        return entries[i].text;
    }

    public long getValue() {
        return entries[getIndex()].size;
    }

}

File: aat-android/src/main/kotlin/ch/bailu/aat/preferences/map/SolidTrimSize.kt
package ch.bailu.aat.preferences.map

import ch.bailu.aat_lib.preferences.SolidIndexList
import ch.bailu.aat_lib.preferences.StorageInterface
import ch.bailu.aat_lib.resources.Res
import ch.bailu.aat_lib.util.MemSize

/**
 * TODO: move to lib
 */
class SolidTrimSize(storageInterface: StorageInterface) : SolidIndexList(
    storageInterface, SolidTrimSize::class.java.simpleName
) {

    private class Entry(val value: Long) {
        val text: String = MemSize.describe(StringBuilder(), value)
    }

    override fun length(): Int {
        return entries.size
    }

    override fun getLabel(): String {
        return Res.str().p_trim_size()
    }

    fun getValue(): Long {
        return entries[index].value
    }

    public override fun getValueAsString(index: Int): String {
        return entries[index].text
    }

    companion object {
        private val entries = arrayOf(
            Entry(16L * MemSize.GB),
            Entry(8L * MemSize.GB),
            Entry(4L * MemSize.GB),
            Entry(2L * MemSize.GB),
            Entry(1L * MemSize.GB),
            Entry(500L * MemSize.MB),
            Entry(200L * MemSize.MB),
            Entry(100L * MemSize.MB),
            Entry(50L * MemSize.MB)
        )
    }
}
